### PR TITLE
feat: add watcher

### PR DESCRIPTION
**Pull request by underfin**

### CODE REVIEW

**Comment by underfin** [comment]
* **#2187** <a href="https://app.graphite.dev/github/pr/rolldown/rolldown/2187?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#2186** <a href="https://app.graphite.dev/github/pr/rolldown/rolldown/2186?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#2185** <a href="https://app.graphite.dev/github/pr/rolldown/rolldown/2185?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* **#2184** <a href="https://app.graphite.dev/github/pr/rolldown/rolldown/2184?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* `hmr-poc`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @underfin and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

File: examples/react-hmr/rolldown.config.js [hunk 1]
```diff
@@ -21,4 +21,5 @@ export default defineConfig({
   output: {
     format: 'app',
   },
+  dev: true,
```
Comment (7086cmd) [inline examples/react-hmr/rolldown.config.js:24]: Is it better to use `hmr` or `watch`? The `dev` option is somewhat ambiguous IMO.

Comment (7086cmd) [reply examples/react-hmr/rolldown.config.js:24]: If intended for a small HTTP server, I would prefer the `serve` option like esbuild: https://esbuild.github.io/api/#serve.

Comment (underfin) [reply examples/react-hmr/rolldown.config.js:24]: It's only a simple option to show case. We need to review the API if we merge it into main branch.

**Pull request merged**